code/_onclick/click.dm

```diff
@@ -402,4 +403,36 @@ GLOBAL_LIST_INIT(click_catchers, create_click_catcher())
 
 /obj/screen/click_catcher/proc/resolve(var/mob/user)
 	var/turf/T = screen_loc2turf(screen_loc, get_turf(user))
-	return T
\ No newline at end of file
+	return T
+
+/mob/living/carbon/human/proc/absolute_grab(mob/living/carbon/human/T)
+	if(!ishuman(T))
+		return
+	if(stat || paralysis || stunned || weakened || lying || restrained() || buckled)
+		to_chat(src, "You cannot leap in your current state.")
+		return
+	if(l_hand && r_hand)
+		to_chat(src, SPAN_DANGER("You need to have one hand free to grab someone."))
+		return
+
+	if(!T || !src || src.stat)
+		return
+	if(get_dist(get_turf(T), get_turf(src)) != 2)
+		return
+	if(last_special > world.time)
+		return
+	last_special = world.time + 75
+	status_flags |= LEAPING
+	src.visible_message(SPAN_DANGER("\The [src] leaps at [T]!"))
+	src.throw_at(get_step(get_turf(T),get_turf(src)), 4, 1, src)
+	playsound(src.loc, 'sound/voice/shriek1.ogg', 50, 1)
+	sleep(5)
+	if(status_flags & LEAPING)
+		status_flags &= ~LEAPING
+
+		if(!src.Adjacent(T))
+			to_chat(src, SPAN_WARNING("You miss!"))
+			Weaken(3)
+			return
+		T.Weaken(1)
```

Comment:
This Weaken may be questionable balance wise, but other than that i see nothing wrong.